pallets/connectors/src/lib.rs

```diff
@@ -378,38 +412,36 @@ pub mod pallet {
 		#[pallet::call_index(5)]
 		pub fn update_member(
 			origin: OriginFor<T>,
-			domain_address: DomainAddress,
 			pool_id: PoolIdOf<T>,
 			tranche_id: TrancheIdOf<T>,
+			domain_address: DomainAddress,
 			valid_until: Moment,
 		) -> DispatchResult {
 			let who = ensure_signed(origin.clone())?;
 
-			// Check that the origin is allowed to add other members
 			ensure!(
-				T::Permission::has(
-					PermissionScope::Pool(pool_id),
-					who.clone(),
-					Role::PoolRole(PoolRole::InvestorAdmin)
-				),
-				BadOrigin
+				T::PoolInspect::pool_exists(pool_id),
+				Error::<T>::PoolNotFound
+			);
+			ensure!(
+				T::PoolInspect::tranche_exists(pool_id, tranche_id),
+				Error::<T>::TrancheNotFound
+			);
+			ensure!(
+				valid_until > Self::now(),
+				Error::<T>::InvalidTrancheInvestorValidity
 			);
 
-			// Now add the destination address as a TrancheInvestor of the given tranche if
-			// not already one. This check is necessary shall a user have called
-			// `update_member` already but the call has failed on the EVM side and needs to
-			// be retried.
-			if !T::Permission::has(
-				PermissionScope::Pool(pool_id),
-				T::AccountConverter::convert(domain_address.clone()),
-				Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, valid_until)),
-			) {
-				T::Permission::add(
+			// Ensure that the destination address has been whitelisted as a TrancheInvestor
+			// beforehand.
+			ensure!(
+				T::Permission::has(
 					PermissionScope::Pool(pool_id),
 					T::AccountConverter::convert(domain_address.clone()),
-					Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, valid_until)),
-				)?;
-			}
+					Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, valid_until))
+				),
+				Error::<T>::DomainNotWhitelisted
```

Comment:
So now this is only testing that the destination domain is whitelisted, no? But we have to ensure that the specific address to which tranche tokens are being sent is an investor and not the destination domain as a whole.

Reply:
Unless if `T::AccountConverter::convert(domain_address.clone()),` considers both the domain and the address; if it doesn't this is wrong and the error name as well 👀

Reply:
> Unless if `T::AccountConverter::convert(domain_address.clone()),` considers both the domain and the address; if it doesn't this is wrong and the error name as well 👀

This [seems to be the case](https://github.com/centrifuge/centrifuge-chain/blob/0d60c08d7326499582b457733bc04f7ceb790868/runtime/common/src/account_conversion.rs#L58): It basically converts to `[...evm_address.bytes(), ...chain_id.bytes(), ...b"EVM]`.

Then it should be better to phrase the error `DomainLocatorNotWhitelisted` or do you have a better suggestion?

Reply:
The _thing_ being whitelisted is an investor's address; it's just that such address is on another chain. So it's not the `DomainLocatorNotWhitelisted` that's not whitelisted but the cross-chain Investor's address. Maybe `InvestorEvmAddressNotWhitelisted`? Or just `InvestorNotWhitelisted`?

Reply:
Went with `InvestorEvmAddressNotWhitelisted` as this is more explicit. Thanks for the explanation!